src/dbnode/storage/index/block.go

```diff
@@ -975,6 +1000,48 @@ func (b *block) EvictMutableSegments() error {
 	return multiErr.FinalError()
 }
 
+func (b *block) NeedsColdMutableSegmentsEvicted() bool {
+	b.RLock()
+	defer b.RUnlock()
+	var anyColdMutableSegmentNeedsEviction bool
+	for _, coldSeg := range b.coldMutableSegments {
+		anyColdMutableSegmentNeedsEviction = anyColdMutableSegmentNeedsEviction || coldSeg.NeedsEviction()
+	}
+	return anyColdMutableSegmentNeedsEviction
```

Comment:
Maybe `return b.state == blockStateSealed && anyColdMutableSegmentNeedsEviction` here?

Otherwise we might think it needs eviction, then we rotate, but when we try to `EvictColdMutableSegments()` it will return an error since it's not sealed?